main.cpp

```diff
@@ -35,39 +30,61 @@ Gnuplot g5("lines");
 void plot_line_gnu(double A, double B, uint16_t size, Gnuplot *g, bool hold_on, string title);
 void plot_gnu(int32_t *array, uint16_t size, Gnuplot *g, bool hold_on, string title);
 
-int main()
+int main(int argc, char *argv[])
 {
-// SELECT WHICH DATABASE YOU WANT:
-//TODO: make this variables for the main function
+
+  //check if too many arguments are passed
+  if (argc != 3) {
+    cout << "Wrong amount of arguments!" << endl;
+    cout << "Usage: ./testing 'stereoboard_nr' 'take_nr'" << endl;
+    cout << "Example: ./testing 1 16" << endl;
+
+    return 0;
+  }
+  cout << "stereoboard " << argv[1] << " with take " << argv[2] << endl;
 
   string configuration_board = "forward_camera";
-  int number_stereoboard = 1;
-  int number_take = 16;	// 3
-  float time_inc = 1/7.; // 1/8
+  int number_stereoboard = atoi(argv[1]);
+  int number_take = atoi(argv[2]);
 
   uint32_t frame_time = 0;
 
   // Find Directories
   stringstream file_directory_images;
+  stringstream file_directory_timing;
   stringstream file_directory_calibration;
   stringstream file_directory_results;
 
   file_directory_images << "stereoboard_database/database_stereoboard_" << number_stereoboard << "/" <<
                         configuration_board << "/take" << number_take << "/%1d.bmp";
+  file_directory_timing << "stereoboard_database/database_stereoboard_" << number_stereoboard << "/" <<
+                        configuration_board << "/take" << number_take << "/timing.dat";
   file_directory_calibration << "stereoboard_database/database_stereoboard_" << number_stereoboard <<
                              "/calibration_data.txt";
   file_directory_results << "stereoboard_database/database_stereoboard_" << number_stereoboard << "/" <<
                          configuration_board << "/take" << number_take << "/result_stereo.csv";
 
-
+  struct cam_state_t cam_state = {NULL};
```

Comment:
This is still not the idea. This will initialise the variables with zeros. You shouldn't initialise variables like this, NULL should only be used with pointers. 

Although this will effectively cause the variables to all be zeros which will in turn make the applied derotation 0, this is not the idea here. If you don't use the derotation you should not send a cam_state. You should explictly disable derotation by passing a NULL pointer. e.g edgeflow_init(128, 94, 0, NULL);